### DISCUSSION TITLE
is there a snap-to-grid mode for component dragging?

### DISCUSSION THREAD

**Discussion by rozek** [GrapesJS/grapesjs#5341] [Q&A]
When dragging a component within the editor, I would like it to snap to a virtual "grid", e.g., with cells 10x10 px large. is there such a feature?

Note: I do _not_ talk about a CSS grid (there is a plugin for such grids) but a restriction of drag positions

**Comment by badiuciprian** [comment]
Hi, i managed to find a way for snapping

const gridSize = 10px;
 editor.on('component:drag:end', (event) => {
      // Calculate the new position based on the grid size
      const element = editor.getSelected();
      const currentPosition = element.getStyle();
      const newX = Math.round(parseInt(currentPosition.left as string) / gridSize) * gridSize;
      const newY = Math.round(parseInt(currentPosition.top as string) / gridSize) * gridSize;
      // Update the element's position
      element.addStyle({
        left: `${newX}px`,
        top: `${newY}px`,
      });
    });
